### PR TITLE
Move Language documentation to the new structure

### PR DESCRIPTION
**Pull request by tmattio**
With #1514 and #1512 being merged (:tada:!) this PR moves the documentation pages in the Language section to the new proposed structure:

- Introduction
- Module System
- Object System
- Data Structures
- Advanced Topics
- Runtime

We have already discussed this, but feedback is welcome if we want to make adjustments before merging 🙂

### CODE REVIEW

File: data/tutorials/language/0lg_01_data_types_and_matching.md
Comment (cuihtlauac) [inline data/tutorials/language/0lg_01_data_types_and_matching.md:1]: This was removed in PR1755

File: data/tutorials/language/2os_00_objects.md
Comment (cuihtlauac) [inline data/tutorials/language/2os_00_objects.md:1]: Do we plan to have several documents in the “Object System” category?

File: data/tutorials/language/4ad_02_pointers.md
Comment (cuihtlauac) [inline data/tutorials/language/4ad_02_pointers.md:1]: We should deprecate or remove this document ASAP

**Review by cuihtlauac — APPROVED** [review]

File: data/tutorials/language/4ad_02_pointers.md
Comment (tmattio) [reply data/tutorials/language/4ad_02_pointers.md:1]: I agree, if there's already a consensus on this, I'll remove it as part of this PR.

File: data/tutorials/language/2os_00_objects.md
Comment (tmattio) [reply data/tutorials/language/2os_00_objects.md:1]: Good question. RWO has two pages on the object system, namely Objects and Classes. It looks to me like there is enough to say for at lease two pages, and likely more. If having a category for it while we don't have the content seems premature, we can also move it to the Advanced Topics section for now.

File: data/tutorials/language/0lg_01_data_types_and_matching.md
Comment (tmattio) [reply data/tutorials/language/0lg_01_data_types_and_matching.md:1]: Should be good with the rebase then

**Comment by cuihtlauac** [comment]
Where would the following topics go?
* Mutability and Imperative Programming
* Labels
* Mutual recursion

**Comment by tmattio** [comment]
> * Mutability and Imperative Programming

Introduction

> * Labels

This one already exists, right? It's in introduction

> * Mutual recursion

I didn't know we wanted a standalone page on this. Can it fit somewhere else? If that's a page, Advanced Topics I suppose.

**Review by sabine — APPROVED** [review]

**Pull request merged**